### PR TITLE
#47 - fixing order of events

### LINKED ISSUE #47: Order of events is not correct
```php
        $agent->webTransaction('Yay', static function () use ($agent) : void {
            $agent->instrument('test', 'foo', static function () use ($agent) : void {
                sleep(1);
                $agent->instrument('test', 'foo2', static function () : void {
                    sleep(1);
                });
            });
            $agent->tagRequest('testtag', '1.23');
        });
```

At the moment, we're getting events in the order more like:

 - StartRequest
 - StartSpan `test/foo2`
 - TagSpan `test/foo2` `stack`
 - StopSpan `test/foo2`
 - StartSpan `test/foo`
 - TagSpan `test/foo` `stack`
 - StopSpan `test/foo`
 - TagRequest `testtag` `1.23`
 - StartSpan `Controller/Yay`
 - TagSpan `Controller/Yay` `stack`
 - StopSpan `Controller/Yay`
 - FinishRequest

The order events is wrong because the `StartSpan test/foo2` refers to the `parent_id` of the `StartSpan Controller/Yay` which doesn't "exist" yet. Also, the order doesn't really make sense due to the order in which the spans are started/stopped etc.

These events should be reordered to their "natural" order, for example:

 - StartRequest
 - StartSpan `Controller/Yay`
 - StartSpan `test/foo`
 - StartSpan `test/foo2`
 - TagSpan `test/foo2` `stack`
 - StopSpan `test/foo2`
 - TagSpan `test/foo` `stack`
 - StopSpan `test/foo`
 - TagRequest `testtag` `1.23`
 - TagSpan `Controller/Yay` `stack`
 - StopSpan `Controller/Yay`
 - FinishRequest

### PR DESCRIPTION
**Pull request by asgrim**
Fixes #47

### CODE REVIEW

File: src/Events/Request/Request.php [hunk 1]
```diff
@@ -36,62 +36,64 @@ public function __construct()
...
-     *
-     * @throws SpanHasNotBeenStarted
      */
     public function stopSpan(?float $overrideTimestamp = null) : void
     {
-        /** @var Span|null $span */
-        $span = array_pop($this->openSpans);
+        $command = $this->currentCommand;
+        if (! $command instanceof Span) {
+            $this->stop($overrideTimestamp);
```
Comment (cschneid) [inline src/Events/Request/Request.php]: In the order of startSpan, req tag, stopSpan, won't currentCommand be the tag, and we'll miss stopping the span?

Comment (asgrim) [reply src/Events/Request/Request.php]: `currentCommand` is practically only a `Request` or `Span`, the `TagRequest` is added directly to `$this->events`

Comment (cschneid) [reply src/Events/Request/Request.php]: Ahh, you're manually managing "currentCommand" in startSpan. Ok.

**Comment by asgrim** [comment]
@cschneid is there anything blocking this merge please? I'd like to rebase #42 once this is merged so the integration test is going to be more effective for the PHP ext being available or not. Thanks!

**Pull request merged**